### ISSUE TITLE
Overflow on FFT normalization for large mesh

### ISSUE THREAD

**Issue by dforero0896** [DifferentiableUniverseInitiative/jaxDecomp#74]
When doing distributed FFTs with large meshes the `_un_normalize_fft` function line 73 on `jaxdecomp/_src/fft_utils.py` does `prod(s)` which overflows for large shapes. A simple cast fixes it though `return jnp.array(prod(map(jnp.float32, s)))` 

Excellent work on the package, thanks!
